src/main/java/com/minecolonies/coremod/entity/citizenhandlers/CitizenHappinessHandler.java

```diff
@@ -0,0 +1,586 @@
+
+    /**
+     * constants for job modifier.
+     */
+    public static final int MAX_DAYS_WITHOUT_JOB = 30;
+    public static final int COMPLAIN_DAYS_WITHOUT_JOB = 7;
+    public static final int DEMANDS_DAYS_WITHOUT_JOB = 14;
+    public static final int MAX_JOB_PENALTY = 5;
+    public static final double JOB_MODIFIER_POSITIVE = 0.5;
+
+    /**
+     * constants for food modifier.
+     */
+    public static final int FOOD_MODIFIER_MAX = -2;
+    public static final int FOOD_MODIFIER_MIN = -1;
+    public static final double FOOD_MODIFIER_POSITIVE = 0.5;
+
+    /**
+     * constants for field modifiers
+     */
+    public static final int FIELD_MAX_DAYS_MODIFIER = 30;
+    public static final double FIELD_MODIFIER_MAX = -0.75;
+    public static final double FIELD_MODIFIER_MIN = -0.15;
+    public static final double FIELD_MODIFIER_POSITIVE = 0.2;
+    public static final double NO_FIELD_MODIFIER = -3.0;
+
+    /**
+     * constants for damage modifiers
+     */
+    public static final int DAMAGE_MODIFIER_MAX = -2;
+    public static final int DAMAGE_MODIFIER_MID = -1;
+    public static final double DAMAGE_MODIFIER_MIN = -0.5;
+
+    /**
+     * constants for happiness min/max and start happines values.
+     */
+    public static final int MAX_HAPPINESS = 10;
+    public static final int MIN_HAPPINESS = 1;
+    public static final int BASE_HAPPINESS = 8;
+
+    /**
+     * constants for no tools.
+     */
+    public static final double NO_TOOL_MODIFIER = -3.0;
+    public static final int NO_TOOLS_MODIFIER = 3;
+    public static final int NO_TOOLS_COMPLAINS_DAYS = 7;
+    public static final int NO_TOOLS_DEMANDS_DAYS = 14;
+    public static final int NO_TOOLS_MAX_DAYS_MODIFIER = 30;
+
+    /**
+     * The citizen assigned to this manager.
+     */
+    private final CitizenData citizen;
+
+    /**
+     * holds the base happiness value.
+     */
+    private double baseHappiness;
+
+    /**
+     * holds the modifier for food.
+     */
+    private double foodModifier;
+
+    /**
+     * holds the modifier for damage.
+     */
+    private double damageModifier;
+
+    /**
+     * holds the modifier for house.
+     */
+    private double houseModifier;
+
+    /**
+     * holds the numbers of days without a house.
+     */
+    private int numberOfDaysWithoutHouse;
+
+    /**
+     * holds the modifier for job.
+     */
+    private double jobModifier;
+
+    /**
+     * holds the numbers of days without a job.
+     */
+    private int numberOfDaysWithoutJob;
+
+    /**
+     * holds the modifier for farms that the farmer can't farm.
+     */
+    private double farmerModifier;
+
+    /**
+     * holds an indicator of the farmer has no fields to farm
+     */
+    private boolean hasNoFields;
+
+    /**
+     * holds an array of all the fields the farmer has assigned to them.
+     */
+    private Map<BlockPos, FieldDataModifier> fieldModifier = new HashMap<BlockPos, FieldDataModifier>();
```

Comment:
I don't get this.  final doesn't appear that it would work on this variable.   I think the check couldn't handle this.fieldModifier  in a couple of methods with the same local variable.